index.js

```diff
@@ -99,30 +99,35 @@ UglifyWriter.prototype.processFile = function(inFile, outFile, relativePath, out
   var src = fs.readFileSync(inFile, 'utf-8');
   var mapName = path.basename(outFile).replace(/\.js$/,'') + '.map';
   var mapDir;
-  var origSourcesContent;
 
   if (this.sourceMapConfig.mapDir) {
     mapDir = path.join(outDir, this.sourceMapConfig.mapDir);
   } else {
     mapDir = path.dirname(path.join(outDir, relativePath));
   }
 
+  var sourceMapOptions = {
+    includeSources: this.options.sourceMapIncludeSources,
+    filename: inFile,
+    url: this.mapURL(mapName)
+  };
+
   var opts = {
-    fromString: true,
-    outSourceMap: this.mapURL(mapName),
-    enableSourcemaps: this.enableSourcemaps()
+    compress: this.options.compress,
+    mangle: this.options.mangle,
+    sourceMap: this.enableSourcemaps() ? sourceMapOptions : false
   };
 
-  if (opts.enableSourcemaps && srcURL.existsIn(src)) {
+  if (this.enableSourcemaps() && srcURL.existsIn(src)) {
     var url = srcURL.getFrom(src);
-    opts.inSourceMap = path.join(path.dirname(inFile), url);
-    origSourcesContent = JSON.parse(fs.readFileSync(opts.inSourceMap)).sourcesContent;
+    opts.sourceMap.content = JSON.parse(fs.readFileSync(path.join(path.dirname(inFile), url)));
```

Comment:
This was somewhat of a guess as to what this code was supposed to do. As far as I can tell, the `origSourcesContent` variable is never used in the current master branch?